Cython/Compiler/Nodes.py

```diff
@@ -2723,6 +2723,16 @@ class DecoratorNode(Node):
     # decorator    NameNode or CallNode or AttributeNode
     child_attrs = ['decorator']
 
+def _get_type_attr(arg, attr):
+    for attrs in (["entry", "type", attr],
+                  ["type", attr]):
```

Comment:
Örks. This helper function suggests to me that there is something wrong with the API, or with the nodes being touched here. This shouldn't be necessary.

Reply:
This is definitely working round an inconsistent existing interface. I'll try to clean it up (and probably submit the clean-up as as a separate small PR).

Reply:
The good news is the mess mostly looks to me one I've both created and worked round in this PR so no need for a separate fix.